plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/ManifestsTable.java

```diff
@@ -137,7 +137,13 @@ private static void writePartitionSummaries(BlockBuilder arrayBlockBuilder, List
 
             BlockBuilder rowBuilder = singleArrayWriter.beginBlockEntry();
             BOOLEAN.writeBoolean(rowBuilder, summary.containsNull());
-            BOOLEAN.writeBoolean(rowBuilder, summary.containsNaN());
+            Boolean containsNan = summary.containsNaN();
+            if (containsNan == null) {
+                rowBuilder.appendNull();
+            }
+            else {
+                BOOLEAN.writeBoolean(rowBuilder, containsNan);
+            }
```

Comment:
Optional may be handy here

```java
            Optional.ofNullable(summary.containsNaN()).ifPresentOrElse(
                    containsNan -> BOOLEAN.writeBoolean(rowBuilder, containsNan),
                    rowBuilder::appendNull);
```

Reply:
I was thinking now also about something similar. A similar change would be beneficial for unboxing other types as well.

Alternatively this could be handled behind the scenes within BOOLEAN if we pass Boolean instead of boolean

Reply:
I'd love to avoid nullable reference types where possible. Quite easy to miss.

The writer being "smarter" means it's easy to introduce mistakes.

Reply:
Hi all

The field is nullable per the spec. It's generally only missing for older data files, so it's usually encountered (and why it's not super reproducible). But it _is_ nullable per the spec. Older tables won't have this field populated and so that is a source of issue.

It is easy to encounter NPEs in my opinion so I typically prefer to handle things with optionals, but it is _usually_ present (if that informs your thinking at all for performance etc) 👍

Reply:
For reference, here is the equivalent code for the scan in Spark: https://github.com/apache/iceberg/blob/68091037944ff7e9de91e7b619f313a8e98c1adc/core/src/main/java/org/apache/iceberg/ManifestsTable.java#L102-L123

Note that the column is nullable: https://github.com/apache/iceberg/blob/68091037944ff7e9de91e7b619f313a8e98c1adc/core/src/main/java/org/apache/iceberg/ManifestsTable.java#L41